// fz-layout/src/components/Layout/docs/Layout__Layout1.docs.js
import React from 'react';
import PropTypes from 'prop-types';
import Container from '../Container';
import Box from '../Box';

export default class Layout__Layout1 extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div style={{ height: '100vh' }}>
        <Container alignBox="column" style={{ background: '#f5f5f5' }}>
          <Box flexible style={{ background: '#f8f8f8' }}>
            <Container
              alignBox="column"
              style={{ background: '#f5f5f5' }}
              hidden={['xs', 'md']}
            >
              <Box hidden={['xs', 'md']}>
                <div style={{ background: '#666', height: '45px' }} />
              </Box>
              <Box flexible>
                <Container alignBox="row" style={{ background: '#f5f5f5' }}>
                  <Box>
                    <div
                      style={{
                        background: '#999',
                        width: '220px',
                        height: '100%'
                      }}
                    />
                  </Box>
                  <Box flexible />
                </Container>
              </Box>
            </Container>
          </Box>
          <Box>
            <div style={{ background: '#f1f1f1', height: '28px' }} />
          </Box>
        </Container>
      </div>
    );
  }
}

Layout__Layout1.docs = {
  componentGroup: 'Template'
};
