// fz-layout/es/components/Layout/docs/Layout__RowAutoHeight.docs.js
import _Object$getPrototypeOf from 'babel-runtime/core-js/object/get-prototype-of';
import _classCallCheck from 'babel-runtime/helpers/classCallCheck';
import _createClass from 'babel-runtime/helpers/createClass';
import _possibleConstructorReturn from 'babel-runtime/helpers/possibleConstructorReturn';
import _inherits from 'babel-runtime/helpers/inherits';
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Grid from '../Grid';
import Col from '../Col';
import Row from '../Row';
function Box() {
  return React.createElement(
    'div',
    { style: { background: '#007fff', minHeight: '1rem' } },
    'asdas'
  );
}

var Layout__RowAutoHeight = function (_Component) {
  _inherits(Layout__RowAutoHeight, _Component);

  function Layout__RowAutoHeight(props) {
    _classCallCheck(this, Layout__RowAutoHeight);

    return _possibleConstructorReturn(this, (Layout__RowAutoHeight.__proto__ || _Object$getPrototypeOf(Layout__RowAutoHeight)).call(this, props));
  }

  _createClass(Layout__RowAutoHeight, [{
    key: 'render',
    value: function render() {
      return React.createElement(
        Grid,
        { style: { width: '100%' } },
        React.createElement(
          Row,
          { style: { height: '100vh', flexDirection: 'column' } },
          React.createElement(
            Col,
            {
              style: {
                background: 'blue'
              }
            },
            'Header'
          ),
          React.createElement(Col, {
            xs: true,
            sm: true,
            md: true,
            lg: true,
            style: {
              background: 'gray'
            }
          })
        )
      );
    }
  }]);

  return Layout__RowAutoHeight;
}(Component);

export default Layout__RowAutoHeight;

if (__DOCS__) {
  Layout__RowAutoHeight.docs = {
    componentGroup: 'Template'
  };
}
