// fz-layout/es/components/Layout/docs/Layout__Layout1.docs.js
import _Object$getPrototypeOf from 'babel-runtime/core-js/object/get-prototype-of';
import _classCallCheck from 'babel-runtime/helpers/classCallCheck';
import _createClass from 'babel-runtime/helpers/createClass';
import _possibleConstructorReturn from 'babel-runtime/helpers/possibleConstructorReturn';
import _inherits from 'babel-runtime/helpers/inherits';
import React from 'react';
import PropTypes from 'prop-types';
import Container from '../Container';
import Box from '../Box';

var Layout__Layout1 = function (_React$Component) {
  _inherits(Layout__Layout1, _React$Component);

  function Layout__Layout1(props) {
    _classCallCheck(this, Layout__Layout1);

    return _possibleConstructorReturn(this, (Layout__Layout1.__proto__ || _Object$getPrototypeOf(Layout__Layout1)).call(this, props));
  }

  _createClass(Layout__Layout1, [{
    key: 'render',
    value: function render() {
      return React.createElement(
        'div',
        { style: { height: '100vh' } },
        React.createElement(
          Container,
          { alignBox: 'column', style: { background: '#f5f5f5' } },
          React.createElement(
            Box,
            { flexible: true, style: { background: '#f8f8f8' } },
            React.createElement(
              Container,
              {
                alignBox: 'column',
                style: { background: '#f5f5f5' },
                hidden: ['xs', 'md']
              },
              React.createElement(
                Box,
                { hidden: ['xs', 'md'] },
                React.createElement('div', { style: { background: '#666', height: '45px' } })
              ),
              React.createElement(
                Box,
                { flexible: true },
                React.createElement(
                  Container,
                  { alignBox: 'row', style: { background: '#f5f5f5' } },
                  React.createElement(
                    Box,
                    null,
                    React.createElement('div', {
                      style: {
                        background: '#999',
                        width: '220px',
                        height: '100%'
                      }
                    })
                  ),
                  React.createElement(Box, { flexible: true })
                )
              )
            )
          ),
          React.createElement(
            Box,
            null,
            React.createElement('div', { style: { background: '#f1f1f1', height: '28px' } })
          )
        )
      );
    }
  }]);

  return Layout__Layout1;
}(React.Component);

export default Layout__Layout1;


Layout__Layout1.docs = {
  componentGroup: 'Template'
};
