// fz-layout/lib/components/Layout/Container.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getContainerProps = getContainerProps;
exports.default = Container;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _Layout = require('./Layout.css');

var _Layout2 = _interopRequireDefault(_Layout);

var _utils = require('../../utils');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var propTypes = {
  reverse: _propTypes2.default.bool,
  className: _propTypes2.default.string,
  alignBox: _propTypes2.default.string,
  children: _propTypes2.default.node,
  tagName: _propTypes2.default.string,
  hidden: _propTypes2.default.array
};
function getContainerClassNames(props) {
  var modificators = [props.className, (0, _utils.getClass)(_Layout2.default, 'row'), (0, _utils.getClass)(_Layout2.default, 'cover')];

  var hidden = props.hidden;
  if (hidden) {
    hidden.forEach(function (key) {
      modificators.push((0, _utils.getClass)(_Layout2.default, 'hidden-screen-' + key));
    });
  }
  if (props.alignBox && props.alignBox == 'row') {
    modificators.push((0, _utils.getClass)(_Layout2.default, 'rowdir'));
  } else {
    modificators.push((0, _utils.getClass)(_Layout2.default, 'coldir'));
  }
  if (props.reverse) {
    modificators.push((0, _utils.getClass)(_Layout2.default, 'reverse'));
  }
  if (props.wrap) {
    modificators.push((0, _utils.getClass)(_Layout2.default, 'wrap'));
  }
  return modificators;
}

function getContainerProps(props) {
  return (0, _utils.createProps)(propTypes, props, getContainerClassNames(props));
}

function Container(props) {
  return _react2.default.createElement(props.tagName || 'div', getContainerProps(props));
}

Container.propTypes = propTypes;
