// fz-layout/es/index.js
import Grid from './components/Layout/Grid';
import Row from './components/Layout/Row';
import Col from './components/Layout/Col';
import Container from './components/Layout/Container';
import Box from './components/Layout/Box';
export { default as React } from 'react';
export { default as ReactDOM } from 'react-dom';

export var Layout = { Grid: Grid, Row: Row, Col: Col, Container: Container, Box: Box };
Layout.docs = {
  componentGroup: 'Template'
};

export { default as Layout__Responsive } from './components/Layout/docs/Layout__default.docs';
export { default as Layout__Offset } from './components/Layout/docs/Layout__Offset.docs';
export { default as Layout__AutoWidth } from './components/Layout/docs/Layout__AutoWidth.docs';
export { default as Layout__Hidden } from './components/Layout/docs/Layout__Hidden.docs';
export { default as Layout__Container } from './components/Layout/docs/Layout__Container.docs';
export { default as Layout__Main } from './components/Layout/docs/Layout__Main.docs';
export { default as Layout__App } from './components/Layout/docs/Layout__App.docs';
export { default as Layout__Lhs } from './components/Layout/docs/Layout__Lhs.docs';
export { default as Layout__Layout1 } from './components/Layout/docs/Layout__Layout1.docs';
export { default as Layout__Layout2 } from './components/Layout/docs/Layout__Layout2.docs';
export { default as Layout__Layout3 } from './components/Layout/docs/Layout__Layout3.docs';
export { default as Layout__Layout4 } from './components/Layout/docs/Layout__Layout4.docs';
export { default as Layout__ThreeColumn } from './components/Layout/docs/Layout__ThreeColumn.docs';
export { default as Layout__FourColumn } from './components/Layout/docs/Layout__FourColumn.docs';
